url-shortner-view/src/FirstView/FirstView.js

```diff
@@ -5,14 +5,15 @@ import { useHistory } from 'react-router-dom';
 
 const FirstView = (props) => {
   const history = useHistory();
+
   const sendUrlHandler = (url) => {
+    props.setSpinner(true);
     if (url !== '') {
-      PostUrl(url)
-        .then((response) => {
-          props.setShortUrl(response.data.short_url);
-          console.log(props.shortUrl);
-        })
-        .then(() => history.push('/shortener'));
+      PostUrl(url).then((response) => {
+        props.setSpinner(false);
+        props.setShortUrl(response.data.short_url);
+        history.push('/shortener');
+      });
```

Comment:
what if the api request fails? and `then` block is not executed? we need to handle the spinner on api fail as well